### PR TITLE
feat: return tasks and notes in getPatientList response

### LINKED ISSUE #488: GetPatientListRequest return tasks and notes
### Describe the feature

The get patient list request should return the tasks and notes for the patient to have full information

### Why is this feature needed?

To display the TaskStatuses properly in mobile

### Additional context

From https://github.com/helpwave/mobile-app/issues/418

### PR DESCRIPTION
**Pull request by PaulKalho**
## Which issues does this pull request close?
closes #488 

## Info

I'm not quite sure if the way I solved mapping the tasks is the right one, happy about feedback.

### CODE REVIEW

File: services/task-svc/internal/patient/patient.go [hunk 1]
```diff
@@ -472,6 +472,29 @@ func (ServiceServer) GetPatientDetails(ctx context.Context, req *pb.GetPatientDe
 	}, nil
 }
 
+func (ServiceServer) mapTasks(tasks []models.Task) []*pb.GetPatientListResponse_Task {
```
Comment (MaxSchaefer) [inline services/task-svc/internal/patient/patient.go:475]:
```suggestion
func mapTasks(tasks []models.Task) []*pb.GetPatientListResponse_Task {
```
Good for reusability, but this function should not be a member of `ServiceServer`. You could also put the function inside `GetPatientList` like `mapTasks := func () {}`.

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
I've got a segmentation violation when calling the endpoint.

```
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | 10:38PM TRC ../libs/hwgorm/logger.go:53 > SELECT "rooms"."name","rooms"."id","rooms"."organization_id","rooms"."ward_id" FROM "rooms" JOIN beds ON rooms.id = beds.room_id JOIN patients ON patients.bed_id = beds.id WHERE patients.organization_id = '3b25c6f5-4705-4074-9fc6-a50c28eba406' AND patients.is_discharged = 0 GROUP BY "rooms"."id" elapsed=13.107919 handler=/proto.services.task_svc.v1.PatientService/GetPatientList organizationID=3b25c6f5-4705-4074-9fc6-a50c28eba406 rowsAffected=4 service=task-svc span= userID=18159713-5d4e-4ad5-94ad-fbb6bb147984 version=
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | panic: runtime error: invalid memory address or nil pointer dereference
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | [signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0xd8ac9e]
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | goroutine 33 [running]:
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | task-svc/internal/patient.ServiceServer.GetPatientList({{}}, {0x1128bd0, 0xc00016c240}, 0xc00047f3e0)
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /build/app/internal/patient/patient.go:538 +0x8de
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | gen/proto/services/task_svc/v1._PatientService_GetPatientList_Handler.func1({0x1128bd0, 0xc00016c240}, {0xef0140?, 0xc00047f3e0})
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /gen/go/proto/services/task_svc/v1/patient_svc_grpc.pb.go:483 +0x78
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | common.validateUnaryInterceptor({0x1128bd0, 0xc00016c240}, {0xef0140, 0xc00047f3e0}, 0x2?, 0xc000010780)
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /libs/common/grpc.go:251 +0x97
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func2.1({0x1128bd0?, 0xc00016c240?}, {0xef0140?, 0xc00047f3e0?})
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/chain.go:48 +0x4f
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | common.authUnaryInterceptor({0x1128bd0, 0xc00047fbc0}, {0xef0140, 0xc00047f3e0}, 0xc0001d4b80, 0xc0005f7d40)
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /libs/common/grpc.go:94 +0x197
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func2.1({0x1128bd0?, 0xc00047fbc0?}, {0xef0140?, 0xc00047f3e0?})
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/chain.go:48 +0x4f
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | common.loggingUnaryInterceptor({0x1128bd0, 0xc00047f3b0}, {0xef0140, 0xc00047f3e0}, 0xc0001d4b80, 0xc0005f7d80)
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /libs/common/grpc.go:294 +0x605
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func2({0x1128bd0, 0xc00047f3b0}, {0xef0140, 0xc00047f3e0}, 0xc0001d4b80, 0xe829e0?)
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/chain.go:53 +0x14e
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | gen/proto/services/task_svc/v1._PatientService_GetPatientList_Handler({0xf85800?, 0x18569c0}, {0x1128bd0, 0xc00047f3b0}, 0xc00014ee00, 0xc0005cbe90)
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /gen/go/proto/services/task_svc/v1/patient_svc_grpc.pb.go:485 +0x138
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | google.golang.org/grpc.(*Server).processUnaryRPC(0xc00002c1e0, {0x112dbd8, 0xc000383a00}, 0xc0003a7d40, 0xc0003b7da0, 0x1816f28, 0x0)
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/google.golang.org/grpc@v1.56.2/server.go:1337 +0xdf3
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | google.golang.org/grpc.(*Server).handleStream(0xc00002c1e0, {0x112dbd8, 0xc000383a00}, 0xc0003a7d40, 0x0)
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/google.golang.org/grpc@v1.56.2/server.go:1714 +0xa36
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | google.golang.org/grpc.(*Server).serveStreams.func1.1()
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/google.golang.org/grpc@v1.56.2/server.go:959 +0x98
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | created by google.golang.org/grpc.(*Server).serveStreams.func1
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/google.golang.org/grpc@v1.56.2/server.go:957 +0x18c
2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | exit status 2
```

**Comment by PaulKalho** [comment]
> I've got a segmentation violation when calling the endpoint.
> 
> ```
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | 10:38PM TRC ../libs/hwgorm/logger.go:53 > SELECT "rooms"."name","rooms"."id","rooms"."organization_id","rooms"."ward_id" FROM "rooms" JOIN beds ON rooms.id = beds.room_id JOIN patients ON patients.bed_id = beds.id WHERE patients.organization_id = '3b25c6f5-4705-4074-9fc6-a50c28eba406' AND patients.is_discharged = 0 GROUP BY "rooms"."id" elapsed=13.107919 handler=/proto.services.task_svc.v1.PatientService/GetPatientList organizationID=3b25c6f5-4705-4074-9fc6-a50c28eba406 rowsAffected=4 service=task-svc span= userID=18159713-5d4e-4ad5-94ad-fbb6bb147984 version=
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | panic: runtime error: invalid memory address or nil pointer dereference
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | [signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0xd8ac9e]
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | goroutine 33 [running]:
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | task-svc/internal/patient.ServiceServer.GetPatientList({{}}, {0x1128bd0, 0xc00016c240}, 0xc00047f3e0)
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /build/app/internal/patient/patient.go:538 +0x8de
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | gen/proto/services/task_svc/v1._PatientService_GetPatientList_Handler.func1({0x1128bd0, 0xc00016c240}, {0xef0140?, 0xc00047f3e0})
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /gen/go/proto/services/task_svc/v1/patient_svc_grpc.pb.go:483 +0x78
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | common.validateUnaryInterceptor({0x1128bd0, 0xc00016c240}, {0xef0140, 0xc00047f3e0}, 0x2?, 0xc000010780)
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /libs/common/grpc.go:251 +0x97
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func2.1({0x1128bd0?, 0xc00016c240?}, {0xef0140?, 0xc00047f3e0?})
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/chain.go:48 +0x4f
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | common.authUnaryInterceptor({0x1128bd0, 0xc00047fbc0}, {0xef0140, 0xc00047f3e0}, 0xc0001d4b80, 0xc0005f7d40)
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /libs/common/grpc.go:94 +0x197
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func2.1({0x1128bd0?, 0xc00047fbc0?}, {0xef0140?, 0xc00047f3e0?})
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/chain.go:48 +0x4f
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | common.loggingUnaryInterceptor({0x1128bd0, 0xc00047f3b0}, {0xef0140, 0xc00047f3e0}, 0xc0001d4b80, 0xc0005f7d80)
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /libs/common/grpc.go:294 +0x605
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func2({0x1128bd0, 0xc00047f3b0}, {0xef0140, 0xc00047f3e0}, 0xc0001d4b80, 0xe829e0?)
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/chain.go:53 +0x14e
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | gen/proto/services/task_svc/v1._PatientService_GetPatientList_Handler({0xf85800?, 0x18569c0}, {0x1128bd0, 0xc00047f3b0}, 0xc00014ee00, 0xc0005cbe90)
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /gen/go/proto/services/task_svc/v1/patient_svc_grpc.pb.go:485 +0x138
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | google.golang.org/grpc.(*Server).processUnaryRPC(0xc00002c1e0, {0x112dbd8, 0xc000383a00}, 0xc0003a7d40, 0xc0003b7da0, 0x1816f28, 0x0)
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/google.golang.org/grpc@v1.56.2/server.go:1337 +0xdf3
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | google.golang.org/grpc.(*Server).handleStream(0xc00002c1e0, {0x112dbd8, 0xc000383a00}, 0xc0003a7d40, 0x0)
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/google.golang.org/grpc@v1.56.2/server.go:1714 +0xa36
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | google.golang.org/grpc.(*Server).serveStreams.func1.1()
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/google.golang.org/grpc@v1.56.2/server.go:959 +0x98
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | created by google.golang.org/grpc.(*Server).serveStreams.func1
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   |     /go/pkg/mod/google.golang.org/grpc@v1.56.2/server.go:957 +0x18c
> 2023-10-31T22:38:10Z app[4d89696a566008] ams [info]app   | exit status 2
> ```

That's strange.  Will look into that

File: services/task-svc/internal/patient/patient.go (hunk 1 above)
Comment (PaulKalho) [reply services/task-svc/internal/patient/patient.go:475]: Jup that was excatly the thing I was struggling with, thank you!

**Comment by use-to** [comment]
@PaulKalho Check your protos first. Take a look at the pipeline error.

File: services/task-svc/internal/patient/patient.go (hunk 1 above)
Comment (PaulKalho) [reply services/task-svc/internal/patient/patient.go]: @MaxSchaefer should be fixed now.

**Review by FoseFx — APPROVED** [review]
Only did a code review, if you've tested it you can merge it.
_We really need to clean up this handler, holy..._

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
Code looks good, but the subtasks are not empty.

**Comment by MaxSchaefer** [comment]
@FoseFx, Paul is on vacation. Can you take a look into this?

**Comment by FoseFx** [comment]
> Code looks good, but the subtasks are not empty.

What do you mean?

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
Thanks to both of you! Looks good. 👍

**Review by MaxSchaefer — APPROVED** [review]

**Pull request merged**